app/jobs/regular/bulk_user_title_update.rb

```diff
@@ -18,6 +19,23 @@ def execute(args)
       end
     end
 
+    # Not called from the job because this needs to happen during the destroy request
+    def self.clear_titles_for_granted_badge(granted_badge_id)
```

Comment:
I think this should just be a method on the `Badge` model rather than putting it in the job, seeing as how the job doesn't use it. Then you don't have to pass the id either, so something like:

```
badge.clear_granted_user_titles
```

The naming of the method is a little tricky because it's like "get rid of all the user titles that have been set based on me (the badge) being granted"

Reply:
Done!